server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -253,8 +257,29 @@ private long removeTabletServerMarkers(Map<UUID,TServerInstance> uidMap,
 
   private long removeFile(Path path) {
     try {
-      if (!useTrash || !fs.moveToTrash(path)) {
-        fs.deleteRecursively(path);
+      switch (removalPolicy) {
+        case TRASH:
+          if (!fs.moveToTrash(path)) {
+            fs.deleteRecursively(path);
+          }
+          break;
+        case ARCHIVE:
+          if (!SimpleGarbageCollector.archiveFile(fs, path)) {
+            fs.deleteRecursively(path);
+          }
+          break;
+        case ARCHIVE_TRASH:
+          if (!SimpleGarbageCollector.archiveFile(fs, path)) {
+            if (!fs.moveToTrash(path)) {
+              fs.deleteRecursively(path);
+            }
+          }
+          break;
+        case DELETE:
+          fs.deleteRecursively(path);
+          break;
+        default:
+          throw new IllegalStateException();
```

Comment:
`new AssertionError("unhandled enum");` is probably more appropriate here than `IllegalStateException`, since it's a programmer error (on us), not state caused by the user.